Source/DafnyCore/AST/TypeDeclarations/NewtypeDecl.cs

```diff
@@ -96,13 +96,21 @@ public Type RhsWithArgument(List<Type> typeArgs) {
   public TopLevelDecl AsTopLevelDecl => this;
   public TypeDeclSynonymInfo SynonymInfo { get; set; }
 
-  public TypeParameter.EqualitySupportValue EqualitySupport {
+  private IndDatatypeDecl.ES equalitySupport = IndDatatypeDecl.ES.NotYetComputed;
```

Comment:
The Rust compiler will also need to have precise equality support, as it's the only backend that needs to distinguish between type-parameter conditional equality and no equality possible.
Hence, it's not only solving a soundness issue, it's also making the compilation to Rust possible.